File: jzoffer1-20/10.cpp
class Solution {
public:
    int rectCover(int number) {
        //n=1,则是1
        //n=2，是2
        if(number==1) return 1;
        if(number==2) return 2;
        //当有n时，第一次放2*1，则剩余f(n-1)
        //第一次放2*2.则剩余f(n-2)
        //return rectCover(number-1)+rectCover(number-2);
        int a=1;
        int b=2;
        int c=0;
        for(int i=3;i<=number;i++){
            c=a+b;
            a=b;
            b=c;
        }
        return c;

    }
};

File: jzoffer1-20/10.js
function jumpFloor(number)
{
    // write code here
    if(number==0) return 0;
    if(number==1) return 1;
        if(number==2) return 2;
        //跳n个台阶的话，从n-1或者n-2台阶开始跳
        //f(n)=f(n-1)+f(n-2)
        //所以是斐波那契
        let a=1;
        let b=2;
        let c=0;
        for(let i=3;i<=number;i++){
            c=a+b;
            a=b;
            b=c;
        }
        return c;
}